store/badger.go

```diff
@@ -49,21 +54,38 @@ func (b *badgerWriteBatch) Destroy() {
 type badgerKV struct {
 	dir string
 	db  *badger.DB
+
+	// Stops gc goroutine when db is closed.
+	closeCh chan struct{}
+	closeWg sync.WaitGroup
 }
 
 func NewBadger(dir string) (*badgerKV, error) { // nolint:golint
-	db, err := badger.Open(badger.DefaultOptions(dir).WithLogger(nullLog{}))
+	if err := os.MkdirAll(dir, 0755); err != nil {
```

Comment:
We should add a test to check that the directory is created, since this is now the expected behavior.